controller/trackerquery.go

```diff
@@ -64,51 +76,57 @@ func (c *TrackerqueryController) Create(ctx *app.CreateTrackerqueryContext) erro
 
 // Show runs the show action.
 func (c *TrackerqueryController) Show(ctx *app.ShowTrackerqueryContext) error {
-	err := application.Transactional(c.db, func(appl application.Application) error {
-		tq, err := appl.TrackerQueries().Load(ctx.Context, ctx.ID)
+	return application.Transactional(c.db, func(appl application.Application) error {
+		trackerquery, err := appl.TrackerQueries().Load(ctx.Context, ctx.ID)
 		if err != nil {
-			cause := errs.Cause(err)
-			switch cause.(type) {
-			case remoteworkitem.NotFoundError:
-				log.Error(ctx, map[string]interface{}{
-					"tracker_id": ctx.ID,
-				}, "tracker query controller not found")
-				return goa.ErrNotFound(err.Error())
-			default:
-				return errs.WithStack(err)
-			}
+			log.Error(ctx, map[string]interface{}{
+				"err":             err,
+				"trackerquery_id": ctx.ID,
+			}, "unable to load the tracker query by ID")
+			return jsonapi.JSONErrorResponse(ctx, err)
+		}
+		result := &app.TrackerQuerySingle{
+			Data: convertTrackerQuery(appl, ctx.Request, *trackerquery),
 		}
-		return ctx.OK(tq)
+		return ctx.OK(result)
 	})
-	if err != nil {
-		return jsonapi.JSONErrorResponse(ctx, err)
-	}
 	return nil
 }
 
 // Update runs the update action.
 func (c *TrackerqueryController) Update(ctx *app.UpdateTrackerqueryContext) error {
-	err := application.Transactional(c.db, func(appl application.Application) error {
-
-		toSave := app.TrackerQuery{
-			ID:            ctx.ID,
-			Query:         ctx.Payload.Query,
-			Schedule:      ctx.Payload.Schedule,
-			TrackerID:     ctx.Payload.TrackerID,
-			Relationships: ctx.Payload.Relationships,
+	_, err := login.ContextIdentity(ctx)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrUnauthorized(err.Error()))
+	}
+	err = validateUpdateTrackerQueryPayload(ctx)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, err)
+	}
+	err = application.Transactional(c.db, func(appl application.Application) error {
+
+		tq, err := appl.TrackerQueries().Load(ctx.Context, *ctx.Payload.Data.ID)
+		if err != nil {
+			return err
+		}
+		if &ctx.Payload.Data.Attributes.Query != nil {
+			tq.Query = ctx.Payload.Data.Attributes.Query
 		}
-		tq, err := appl.TrackerQueries().Save(ctx.Context, toSave)
+		if &ctx.Payload.Data.Attributes.Schedule != nil {
+			tq.Schedule = ctx.Payload.Data.Attributes.Schedule
+		}
+		if &ctx.Payload.Data.Relationships.Tracker.Data.ID != nil {
+			tq.TrackerID = uuid.FromStringOrNil(*ctx.Payload.Data.Relationships.Tracker.Data.ID)
+		}
+		_, err = appl.TrackerQueries().Save(ctx.Context, *tq)
 
 		if err != nil {
-			cause := errs.Cause(err)
-			switch cause.(type) {
-			case remoteworkitem.BadParameterError, remoteworkitem.ConversionError:
-				return goa.ErrBadRequest(err.Error())
-			default:
-				return goa.ErrInternal(err.Error())
-			}
+			return err
```

Comment:
fixed it here 27e8b4a

Reply:
No you didn't. Instead you've called `return jsonapi.JSONErrorResponse(ctx, err)` and that does nothing to the error itself. I wanted you to use `errs.Wrapf(err, "a bit more context on the error")`. With `return jsonapi.JSONErrorResponse(ctx, err)` you only return from `err = application.Transactional(c.db, func(appl application.Application) error {` and after that you call `jsonapi.JSONErrorResponse(ctx, err)` again.

Reply:
Is this okay? 5f2be1b

Reply:
No, because because inside the `err = application.Transactional(c.db, func(appl application.Application) error {` you should return a regular error but outside you're on the JSONAPI level where you should transform the error into a JSONAPI error instead. That's why you should use `return jsonapi.JSONErrorResponse(ctx, err)` on the outside.

Reply:
i return jsonapi error on top level d464d47